semigroupoids.cabal

```diff
@@ -126,11 +127,15 @@ library
   build-depends:
     base                >= 4.9     && < 5,
     base-orphans        >= 0.8.4   && < 1,
+    -- TODO RGS: Bump lower version bound once changes are on Hackage
     bifunctors          >= 5.5.9   && < 6,
     template-haskell    >= 0.2.11,
     transformers        >= 0.5     && < 0.7,
     transformers-compat >= 0.6     && < 0.8
 
+  if !impl(ghc >= 9.6)
+    build-depends: foldable1-classes-compat >= 0.1 && < 0.2
+
   if flag(containers)
     build-depends: containers >= 0.5.7.1 && < 0.7
```

Comment:
Try something like

```
cabal build -w $HOME/.ghcup/bin/ghc-9.6.0.20230128 all -f-containers --allow-newer=template-haskell --allow-newer=base
```

The `-f-containers` is import bit.

But TBH, I don't know why one would disable `containers` flag ever. I'd simply remove it.

Reply:
Ah, I see. After macro expansion, `defined(MIN_VERSION_containers) && HAS_FOLDABLE1_CONTAINERS` becomes:

```c
defined(MIN_VERSION_containers) && MIN_VERSION_containers(0,6,7)
```

But if `MIN_VERSION_containers` isn't defined, then that is a malformed macro. (I had assumed that `define(MIN_VERSION_containers)` would short-circuit if it wasn't defined, but sadly, that is not the case.)

I've pushed a new version which ensures that puts the macros into a well behaved order.

> But TBH, I don't know why one would disable `containers` flag ever. I'd simply remove it.

Yes, I feel the same way. I've opened #133 to track removing this flag.